liblangutil/SourceLocation.h

```diff
@@ -119,4 +119,23 @@ SourceLocation parseSourceLocation(
 /// Stream output for Location (used e.g. in boost exceptions).
 std::ostream& operator<<(std::ostream& _out, SourceLocation const& _location);
 
+
+/**
+ * Alternative, line-column-based representation for source locations.
+ * Both line and column are zero-based.
+ * If used as a range, the second location is considered exclusive.
+ * Negative values are invalid.
+ */
+struct LineColumn
+{
+	/// Line value, can be between zero and number of `\n` characters in the source file.
+	int line = {-1};
+	/// Column value, can be between zero and number of characters in the line (inclusive).
+	int column = {-1};
+
+	LineColumn() = default;
+	explicit LineColumn(int _line, int _column): line(_line), column(_column) {}
```

Comment:
Does it even need a constructor now that arguments match struct fields and are no longer a tuple?

Reply:
Yes, but can we have default values and prevent construction from just a single value?

Reply:
True, that's a good argument for keeping the constructor.